File: 11. Container With Most Water/container_with_most_water.py
def absolute (number):
	return (number * -1) if (number < 0) else (number * 1)
	
def get_volume (height1, height2, width):
	height = (height1) if (height1 < height2) else (height2);
	
	return height * width

def container_points (array):
	len_array = len(array)
	largest_volume = 0
	volume_indexes = []
	
	for a in range(len_array):
		for b in range (len_array - 1, 0, -1):
			#print(str(a) + "\t\t" + str(b) + "\t|\t" + str(array[a]) + "\t" + str(array[b]))
			if (a == b):
				#no volume
				continue
			elif (b < a):
				break
			else:
				#smaller_element = array[a] if (array[a] < array[b]) else array[b]
				interval = absolute(a - b)
				#volume = smaller_element * interval
				volume = get_volume (array[a], array[b], interval)
				
				if (largest_volume < volume):
					largest_volume = volume
					volume_indexes = [a,b]
	
	return volume_indexes
	
height_array = [1,8,6,2,5,4,8,3,7]
height_indices = container_points(height_array)
height1 = height_array[height_indices[0]]
height2 = height_array[height_indices[1]]
interval = absolute(height_indices[0] - height_indices[1]);

print("This is the handy work of Python\n\n");
print(height_indices);
print("The heights given the points are: " + str(height1) + " on one end and " + str(height2) + " on the other end. The difference between the two points is " + str(interval) + " giving a total volume of " + str (get_volume(height1, height2, interval)))

File: 11. Container With Most Water/container_with_most_water.js
function absolute (number){
	return (number < 0) ? (number * -1) : (number * 1);
}

function getVolume (height1, height2, width){
	let height = (height1 < height2) ? (height1) : (height2);
	
	return (height * width);
}

function containerPoints (list){
	let lenList = list.length;
	let largestVolume = 0;
	let volumeIndexes = [];
	
	for (let a = 0; a < lenList; a++){
		for (let b = (lenList - 1); b > 0; b--){
			if (a == b){
				continue;
			} else if (b < a){
				break;
			} else {
				//let smallerElement = (list[a] < list[b]) ? (list[a]) : (list[b]);
				let interval = absolute(a - b);
				//let volume = smallerElement * interval;
				let volume = getVolume(list[a], list[b], interval);
				
				if (largestVolume < volume){
					largestVolume = volume
					volumeIndexes = [a,b];
				}
			}
		}
	}
	
	return (volumeIndexes);
}

let heightList = [1,8,6,2,5,4,8,3,7];
let heightIndices = containerPoints(heightList);
let height1 = heightList[heightIndices[0]]
let height2 = heightList[heightIndices[1]]
let interval = absolute(heightIndices[0] - heightIndices[1]);

console.log("This is the handy work of JavaScript\n\n");
console.log(heightIndices);
console.log("The heights given the points are: " + String(height1) + " on one end and " + String(height2) + " on the other end. The difference between the two points is " + String(interval) + " giving a total volume of " + String(getVolume(height1, height2, interval)) + "\n\n");
